server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

```diff
@@ -87,19 +88,28 @@ public void execute(SearchContext context) {
             LOGGER.trace("{}", new SearchContextSourcePrinter(context));
         }
 
-        Map<String, Set<String>> storedToRequestedFields = new HashMap<>();
-        FieldsVisitor fieldsVisitor = createStoredFieldsVisitor(context, storedToRequestedFields);
+        if (context.docIdsToLoadSize() == 0) {
+            // no individual hits to process, so we shortcut
```

Comment:
We were checking in a couple of subphases for things like "how many docs are we executing against" or "is this a suggest-only response".  This pulls this check up to the top level, which also means we don't need to include this information in the FetchContext.